client/src/mock/api/expense/expenses.ts

```diff
@@ -1,67 +1,109 @@
+		const recurrenceType = recurrenceTypes.find(
+			(recurrenceType) => recurrenceType.id === expense.recurrence_type_id
+		);
+		const currencyType = currencyTypes.find(
+			(currencyType) => currencyType.id === expense.currency_type_id
+		);
+		return {
+			id: expense.id,
+			name: expense.name,
+			description: expense.description,
+			recurrenceType: recurrenceType ? recurrenceType : { id: 0, name: '', perYear: 0 },
+			currencyType: currencyType ? currencyType : { id: 0, abbreviation: '', name: '' },
+			value: expense.value,
+		};
+	});
+
+	return response(200, data) as GetPredefinedExpensesResponse;
+};
+
 export const mockGetExpenses = (data: GetExpensesRequestData): GetExpensesResponse => {
-	loadTransactions();
+	loadExpenses();
 
 	if (!data.userId) {
 		return response(400, { expenses: [] }) as GetExpensesResponse;
 	}
 
-	const userExpenses = expenses
-		.filter((expense) => expense.user_id === data.userId)
-		.map((expense) => {
-			return {
-				id: expense.id,
-				name: expense.name,
-				description: expense.description,
-				recurrenceType: recurrenceTypes.find(
-					(recurrenceType) => recurrenceType.id === expense.recurrence_type_id
-				),
-				currencyType: currencyTypes.find(
-					(currencyType) => currencyType.id === expense.currency_type_id
-				),
-				predefinedExpenseId: expense.predefined_expense_id,
-				startDate: expense.startDate,
-				value: expense.value,
-				userId: expense.user_id,
-				transactions: transactions?.filter((transaction) => expense.id === transaction.expenseId),
-			} as Expense;
-		});
+	const userExpenses = expenses?.filter((expense) => expense.userId === data.userId);
+	userExpenses?.forEach((expense) => {
+		const expenseTransactions = transactions?.filter(
+			(transaction) => transaction.expenseId === expense.id
+		);
+		if (expenseTransactions) {
+			expense.transactions = expenseTransactions;
+		}
+	});
 	return response(200, { expenses: userExpenses }) as GetExpensesResponse;
 };
 
 export const mockGetCurrencyTypes = (): GetCurrencyTypesResponse => {
 	return response(200, currencyTypes) as GetCurrencyTypesResponse;
 };
 
+export const mockGetRecurrenceTypes = (): GetRecurrenceTypesResponse => {
+	return response(200, recurrenceTypes) as GetRecurrenceTypesResponse;
+};
+
+export const mockNewExpense = (data: NewExpenseRequestData): NewExpenseResponse => {
+	loadExpenses();
+	const newExpense = data.newExpense;
+	const expenseId = Math.floor(Math.random() * 100000);
+	const recurrenceType = recurrenceTypes.find(
+		(recurrenceType) => recurrenceType.id === newExpense.recurrenceTypeId
+	);
+	const currencyType = currencyTypes.find(
+		(currencyType) => currencyType.id === newExpense.currencyTypeId
+	);
+	const expense: Expense = {
+		id: expenseId,
+		name: newExpense.name,
+		description: newExpense.description,
+		recurrenceType: recurrenceType ? recurrenceType : { id: 0, name: '', perYear: 0 },
+		currencyType: currencyType ? currencyType : { id: 0, abbreviation: '', name: '' },
```

Comment:
Same as above.
Although I would probably create a function for this mapping, which takes an expanse, a recurrenceType ID and a currencyType ID (so it could include finds as well). You could reuse it at least 2 times.

Reply:
I don't really want to pour any more hours into this mock tbh.